### PR TITLE
New resource: aws_ssoadmin_customer_managed_policy_attachment

### LINKED ISSUE #25904: Support for AWS SSO permission sets with customer managed policies
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Add support for using customer managed policies (CMP) with AWS SSO permission sets.  This feature was released on July 19, 2022, [blog post for release](https://aws.amazon.com/about-aws/whats-new/2022/07/aws-single-sign-on-aws-sso-aws-identity-access-management-iam-customer-managed-policies-cmps/), [SSO documentation](https://docs.aws.amazon.com/singlesignon/latest/userguide/howtocmp.html).

### New or Affected Resource(s)

* aws_XXXXX

### Potential Terraform Configuration

```hcl
resource "aws_ssoadmin_permission_set_customer_managed_policy" "example" {
  policy_arn         = "arn:aws:iam::000000000000:policy/my-policy"
  instance_arn       = aws_ssoadmin_permission_set.example.instance_arn
  permission_set_arn = aws_ssoadmin_permission_set.example.arn
}
```

### References

* #0000

### LINKED ISSUE #26068: AWS SSO - Support for Customer managed IAM policies with permission sets
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Amazon released new feature for SSO that allows customer managed keys to be assigned to permission sets to provide fine grained access. 
https://aws.amazon.com/about-aws/whats-new/2022/07/aws-single-sign-on-aws-sso-aws-identity-access-management-iam-customer-managed-policies-cmps/

### New or Affected Resource(s)
This feature will require new resource block to support adding custom customer managed policies to permission set..

### References
https://aws.amazon.com/about-aws/whats-new/2022/07/aws-single-sign-on-aws-sso-aws-identity-access-management-iam-customer-managed-policies-cmps/

### PR DESCRIPTION
**Pull request by oakbramble**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #25904.
Closes #26068.

This is my first contribution and first time working with Go, so thank you in advance for your feedback.
I have added `aws_ssoadmin_customer_managed_policy_attachment` and manually tested that it is working, creating the following resource:
```
resource "aws_ssoadmin_customer_managed_policy_attachment" "example" {
  customer_managed_policy_name         = aws_iam_policy.example.name
  customer_managed_policy_path         = "/" # optional
  instance_arn       = tolist(data.aws_ssoadmin_instances.example.arns)[0]
  permission_set_arn = aws_ssoadmin_permission_set.example.arn
}
```
I based the acceptance tests on `managed_policy_attachment_test.go`. They are passing (see below), however there are a couple of areas I am unsure about:
- I have used some random names in the test resources being created, to avoid conflicts, however am unsure if I'm doing this properly?
- Not all of the resources are being destroyed after the tests, I assume this could be related to the above and `testAccCheckCustomerManagedPolicyAttachmentDestroy`?

***Update: The tests are working and all resources being destroyed properly. I have marked the PR ready for review***

Thanks for bearing with me as I get used to the repo and I hope this PR is helpful.

Output from acceptance testing:

```
$ make testacc TESTS=TestAccSSOAdminCustomerManagedPolicyAttachment PKG=ssoadmin

==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/ssoadmin/... -v -count 1 -parallel 20 -run='TestAccSSOAdminCustomerManagedPolicyAttachment'  -timeout 5m
=== RUN   TestAccSSOAdminCustomerManagedPolicyAttachment_basic
=== PAUSE TestAccSSOAdminCustomerManagedPolicyAttachment_basic
=== RUN   TestAccSSOAdminCustomerManagedPolicyAttachment_forceNew
=== PAUSE TestAccSSOAdminCustomerManagedPolicyAttachment_forceNew
=== RUN   TestAccSSOAdminCustomerManagedPolicyAttachment_disappears
--- PASS: TestAccSSOAdminCustomerManagedPolicyAttachment_disappears (27.91s)
=== RUN   TestAccSSOAdminCustomerManagedPolicyAttachment_Disappears_permissionSet
--- PASS: TestAccSSOAdminCustomerManagedPolicyAttachment_Disappears_permissionSet (22.14s)
=== RUN   TestAccSSOAdminCustomerManagedPolicyAttachment_multipleManagedPolicies
=== PAUSE TestAccSSOAdminCustomerManagedPolicyAttachment_multipleManagedPolicies
=== CONT  TestAccSSOAdminCustomerManagedPolicyAttachment_basic
=== CONT  TestAccSSOAdminCustomerManagedPolicyAttachment_multipleManagedPolicies
=== CONT  TestAccSSOAdminCustomerManagedPolicyAttachment_forceNew
--- PASS: TestAccSSOAdminCustomerManagedPolicyAttachment_basic (31.19s)
--- PASS: TestAccSSOAdminCustomerManagedPolicyAttachment_multipleManagedPolicies (51.96s)
--- PASS: TestAccSSOAdminCustomerManagedPolicyAttachment_forceNew (56.06s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/ssoadmin	106.228s

```

### CODE REVIEW

**Comment by oakbramble** [comment]
Ok, I had the tests working, however now I am having trouble attaching multiple policies, both when testing manually and in the acc tests. 
```
  Error: error attaching Customer Managed Policy to SSO Permission Set (arn:aws:sso:::permissionSet/ssoins-68049fb1eee8a70d/ps-69539f9a26bf4698): ConflictException: Conflicting operation occurred on Permission Set with Id: ps-69539f9a26bf4698 and SSOInstanceArn: arn:aws:sso:::instance/ssoins-68049fb1eee8a70d.
```
It has a conflict when adding or deleting more than one `aws_ssoadmin_customer_managed_policy_attachment` resource at the same time. Adding them sequentially is ok. Any advice on how I can make sure these calls are handled properly, so they do not cause conflict, would be much apprecated.

**Comment by csanders-git** [comment]
Same thing occurs with detaching multiple CMPolicies.

```
│ Error: error detaching Customer Managed Policy (potato2) from SSO Permission Set (arn:aws:sso:::permissionSet/ssoins-722321efc6569e0f/ps-e49642a3f1f3d8ca): ConflictException: Conflicting operation occurred on Permission Set with Id: ps-e49642a3f1f3d8ca and SSOInstanceArn: arn:aws:sso:::instance/ssoins-722321efc6569e0f.
```

**Comment by oakbramble** [comment]
Thanks @csanders-git for taking a look and providing feedback. I have added retry conditions to the create and delete attachment functions (I know you mention a nicer solution, but this is what I could get working). I find that the tests are now consistently passing and with manual testing I can attach and detach multiple customer managed policies at the same time. 

(Side note: I do wonder why this conflict doesn't seem to be an issue for the AWS managed policy attachments, unless I am missing a different retry logic around them?)

Please let me know if there is anything else you see issue with, or if you do not find that this solution for multiple attachments is working.

**Comment by nomeelnoj** [comment]
The [AWS CLI](https://docs.aws.amazon.com/cli/latest/reference/sso-admin/attach-customer-managed-policy-reference-to-permission-set.html) and [API docs](https://docs.aws.amazon.com/singlesignon/latest/APIReference/API_AttachCustomerManagedPolicyReferenceToPermissionSet.html) both use a config block for the `CustomerManagedPolicyReference`.  We should do the same here (easy to extend in the future if/when they allow searching for the policy by tags, etc.)

I also think we should call it what it is, which is a reference to a policy.  Please also make sure in the docs it is clear that in order for this to work, the customer managed policy has to first be present in the account that is getting assigned to the permission set.

```
resource "aws_ssoadmin_customer_managed_policy_attachment" "example" {
  customer_managed_policy_reference { # This should end in `reference`, as well as be a config block
    name         = aws_iam_policy.example.name
    path         = aws_iam_policy.example.path # Optional
  }
  instance_arn       = tolist(data.aws_ssoadmin_instances.example.arns)[0]
  permission_set_arn = aws_ssoadmin_permission_set.example.arn
}
```
This would be more inline with the AWS API docs and be easier to understand and support.

Also, doing it with a config block allows for better terraform logic with the ability to use dynamics to find and attach policies only for permission sets that are deployed in accounts where the policies exist.

**Comment by paulschwarzenberger** [comment]
> (Side note: I do wonder why this conflict doesn't seem to be an issue for the AWS managed policy attachments, unless I am missing a different retry logic around them?)

@oakbramble thank you so much for working on this, I will be using the new feature as soon as it's released!

I believe I've seen the equivalent issue with AWS managed policy attachments, resulting in timeouts and failure to apply when trying to attach managed policies to large numbers of accounts and groups at the same time via a module.  I worked around the issue by adding an otherwise unnecessary `for_each` statement to the resource.

**Comment by oakbramble** [comment]
Thanks for the feedback @nomeelnoj . I had been thinking from a user perspective it would be more readable without a config block, particularly when `path` will likely not be used so often, but I understand the reasoning to make it more like the API.

I have not had much time to look at this today, however have reconfigured to use the config block and tested manually that this is working (basic, multiple policies, force new and delete). I have not sorted the tests or docs yet, but will look at them tomorrow. In the meantime, I have pushed the WIP, just in case there are any further tips forthcoming :smile:   (I am getting to grips with the repo/ Go, thank you for bearing with me.)

@paulschwarzenberger  thanks for the info about the managed policy attachments - I'll watch out for that :)

File: website/docs/r/ssoadmin_customer_managed_policy_attachment.html.markdown
Comment (ColinHarrington) [inline website/docs/r/ssoadmin_customer_managed_policy_attachment.html.markdown:56]: The documentation needs an update to reflect the `customer_managed_policy_reference` form instead of the name/path breakdown

File: internal/provider/provider.go
Comment (ColinHarrington) [inline internal/provider/provider.go:2026]: It looks like there is a convention of putting these resources in alphabetical sort order, can you move this one above the `aws_ssoadmin_managed_policy_attachment`

File: internal/service/ssoadmin/customer_managed_policy_attachment.go
Comment (oakbramble) [reply internal/service/ssoadmin/customer_managed_policy_attachment.go:60]: Sure, I'd based this on a different resource that also has a similar path element, but makes sense to not have the default. I'll look into it.

**Comment by oakbramble** [comment]
Following @nomeelnoj 's comments, there is now a functioning `customer_managed_policy_reference` config block and the tests have been updated to reflect this.

@ColinHarrington thanks for the review. I will look at removing the path default and updating documentation tomorrow morning.

File: internal/service/ssoadmin/customer_managed_policy_attachment.go
Comment (oakbramble) [reply internal/service/ssoadmin/customer_managed_policy_attachment.go:60]: Looking at this, the path needs to be defined at some point, even if it is the default, as it is required to ensure consistency in the ID format. Once it has been created, the ID is used to find the resource, so the path will be there already for all other actions. Unless this is causing an actual bug in the resource, I'm not sure refactoring to not include the default path in the create API call is worth the extra work and delay.

Comment (oakbramble) [reply internal/service/ssoadmin/customer_managed_policy_attachment.go:60]: Also, the `aws_iam_policy` resource uses the default value for `path` in the same way: https://github.com/hashicorp/terraform-provider-aws/blob/main/internal/service/iam/policy.go

**Comment by oakbramble** [comment]
Ok, docs have been updated and all looks to be in order. To update on the initial post, this is what the final resource looks like: 
```
resource "aws_ssoadmin_customer_managed_policy_attachment" "example" {
  instance_arn       = tolist(data.aws_ssoadmin_instances.example.arns)[0]
  permission_set_arn = aws_ssoadmin_permission_set.example.arn
  customer_managed_policy_reference {
    name       = aws_iam_policy.example.name
    path         = "/" # optional
  }
}
```
This  is the current output from the acceptance testing:
```
$ make testacc TESTS=TestAccSSOAdminCustomerManagedPolicyAttachment PKG=ssoadmin

==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/ssoadmin/... -v -count 1 -parallel 20 -run='TestAccSSOAdminCustomerManagedPolicyAttachment'  -timeout 180m
=== RUN   TestAccSSOAdminCustomerManagedPolicyAttachment_basic
=== PAUSE TestAccSSOAdminCustomerManagedPolicyAttachment_basic
=== RUN   TestAccSSOAdminCustomerManagedPolicyAttachment_forceNew
=== PAUSE TestAccSSOAdminCustomerManagedPolicyAttachment_forceNew
=== RUN   TestAccSSOAdminCustomerManagedPolicyAttachment_disappears
--- PASS: TestAccSSOAdminCustomerManagedPolicyAttachment_disappears (22.10s)
=== RUN   TestAccSSOAdminCustomerManagedPolicyAttachment_Disappears_permissionSet
--- PASS: TestAccSSOAdminCustomerManagedPolicyAttachment_Disappears_permissionSet (49.79s)
=== RUN   TestAccSSOAdminCustomerManagedPolicyAttachment_multipleManagedPolicies
=== PAUSE TestAccSSOAdminCustomerManagedPolicyAttachment_multipleManagedPolicies
=== CONT  TestAccSSOAdminCustomerManagedPolicyAttachment_basic
=== CONT  TestAccSSOAdminCustomerManagedPolicyAttachment_multipleManagedPolicies
=== CONT  TestAccSSOAdminCustomerManagedPolicyAttachment_forceNew
--- PASS: TestAccSSOAdminCustomerManagedPolicyAttachment_basic (24.02s)
--- PASS: TestAccSSOAdminCustomerManagedPolicyAttachment_multipleManagedPolicies (39.16s)
--- PASS: TestAccSSOAdminCustomerManagedPolicyAttachment_forceNew (42.87s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/ssoadmin	114.810s

```

**Comment by oakbramble** [comment]
After the pipeline tests were run last week, I had some errors to correct (sorry I did not see all these extra tests in the make file beforehand). Below is summary of the corrections I have made, based on [these](https://github.com/hashicorp/terraform-provider-aws/pull/25915/checks?check_run_id=7560407126) errors.

- Remove a trailing whitespace in documentation ([here](https://github.com/hashicorp/terraform-provider-aws/pull/25915/commits/49e707ffa7e6a904fcb9e893da6d4a6037e00ef1))
- Switch from depreciated `acctest.ProviderFactories` to `acctest.ProtoV5ProviderFactories` ([here](https://github.com/hashicorp/terraform-provider-aws/pull/25915/commits/85b7d89871514c7a5e83a5597e8bc218a6ee96dd))
- Add `TimedOut` to a Retry and remove unnecessary type conversions ([here](https://github.com/hashicorp/terraform-provider-aws/pull/25915/commits/509a6bbd8f3d96c3bac7a87c25bdf78965235a23))
- Correct import statement in docs ([here](https://github.com/hashicorp/terraform-provider-aws/pull/25915/commits/bb694781ded94bd24bb6f61e6ada4ac9f88771da))
- Update tf code in tests to pass terrafmt ([here](https://github.com/hashicorp/terraform-provider-aws/pull/25915/commits/911f870af1101e19af659a6f55c979dfc311a2d0))

In addition to the acc tests, the following are now also passing when I run locally:
- `make semgrep`
- `terrafmt`
- `make fmt`
- `make importlint`
- `make golang-ci-lint`
- `make docs-lint`

I'm having trouble _running_ `make providerlint` however the provider lint tests did pass in the pipeline already.

Please let me know if there are any more concerns, or if there's anything else I should be doing to get this merged. 

(I know the branch doesn't match the naming convention, but I didn't see this until it was too late; will do better next time.)

**Comment by csanders-git** [comment]
I have retested the latest version of the PR, it works flawlessly. 

The contribution guidelines indicate that comments are also an important part of prioritizing PRs, so here we go:

Let me take a stab at discussing why this is so critical to many orgs. AWS-SSO allows for a proper SCIM based integration with IdP's such as Okta. This is only available in a hacky manner via the IAM SAML integration because it doesn't speak SCIM AND it doesn't provide an OIDC Authorization Server for accessing credentials via the CLI which leads to terrible command line tools to assume roles via CLI. 
The most recent change corrects a long standing issue with AWS-SSO that effectively made it incompatible with how IAM is supposed to be used in a secure way. Traditionally, IAM encourages organizations to make fine grained reusable policies (customer managed policies) to augment the preset AWS managed policies and provide fine-grained, least priv access.  
AWS-SSO broke this model because it (previously) ONLY supported inline (considered to be bad security practice) and AWS Managed Policies. Not only did this make it difficult for most organizations that were properly using IAM to migrate, it also meant there was a functional limit on the size of policies that was different from IAM (i.e one inline policy can only hold N chars where N is less than N*10 customer managed policies).
AWS fixed this issue recently by allowing (this admittedly sorta hacky solution) to let users associate existing customer managed policies with a permission set (translation between SCIM groups/users and Roles). 

This change, that @oakbramble so quickly added support for, frees up many organizations to migrate to AWS and GREATLY improve their security posture.

**Comment by tom-henderson** [comment]
Going to add another vote here to help prioritise this. Naturally AWS released this feature about 2 weeks after we complete our migration to SSO managed inline policies, but this would still be very useful to us.

**Comment by josemdav** [comment]
In addition to what @csanders-git mentions, another business need that support for customer managed policies in permission sets is providing flexibility to application owners / developers to directly manage the permissions at the AWS account level without having to modify the centrally defined permission set. For organizations in which an internal shared responsibility model is implemented, this adds agility.

**Comment by bgupta** [comment]
Please note that many AWS customers were waiting for Amazon to release this feature, and many are now blocked from deploying if they are Terraform customers, even if they don't realize it yet. The inability to manage AWS permissions in Okta via Terraform has been a long-standing pain point, and we're very much looking forward to seeing this feature implemented!

**Comment by c0mput3rj0n3s** [comment]
This is a huge deal for us as we're in the midst of a migration from IAM users to AWS SSO in our org. We can't be fine-grained with the SSO inline policies, and we easily bust the 10KB limits if we try, and this feature appearing when it did would have been excellent had the Terraform provider been updated to support it. This would solve _so many problems_ that I'd say it's desperately needed.

**Comment by raizyr** [comment]
I am in the same boat as @c0mput3rj0n3s.  We are trying to migrate from IAM to AWS SSO and have similar concerns.  This terraform resource is badly needed.

**Comment by nocode99** [comment]
A hacky workaround is to use a null-resource and use the AWS CLI. The caveat is if you need to delete something, you need to do it manually. 

```
resource "null_resource" "cmp" {
  for_each = local.cmp_aws_policies

  triggers = {
    new_policies = each.value
  }

  provisioner "local-exec" {
    command = "aws sso-admin attach-customer-managed-policy-reference-to-permission-set --instance-arn ${local.aws_ssoadmin_instance_arn} --permission-set-arn ${aws_ssoadmin_permission_set.pm.arn} --customer-managed-policy-reference Name=${each.value}"
  }
}
```

**FWIW** - AWS does **NOT** validate whether the CMP's exist or not.

**Comment by altosys** [comment]
Is there anything preventing the merge of this PR as it is now? We along with many others are in rather urgent need of getting this into the provider and I'm personally trying to fend off implementing custom permissions manually via the management console as best I can in hoping that this will be pushed out soon.

**Comment by oakbramble** [comment]
Thanks @csanders-git for kicking off the comments to get this noticed. If there's anything I should or could be doing to get this noticed, or anything I have missed then I am absolutely here to do that. I know the branch name isn't in the convention (as I already mentioned, I missed this when starting the PR) but as far as I'm aware, there's nothing I can do about that.

To add to the comments above, I am also blocked at work by this resource not being available. I have been waiting for this feature to come to SSO for months as the inline policy limit is hugely restrictive when working with fine-grained permissions (and is just generally not a great way to do things). Since starting this PR I have been working on refactoring the Terraform for our internal platform and these policy attachments are now the only thing remaining. It would be a real shame to have to go and do all of this work manually and untracked in the console whilst waiting for this to become available in the TF provider.

**Comment by oakbramble** [comment]
Ok, the tests were triggered. The only check that failed was the markdown links, but that looks to be because the link to the contribution pages has changed in the past weeks and this has already been fixed throughout the provider. I can't see anything failing that is actually to do with the PR. (Please correct me if I am missing something.)

If anyone has advice on what I should do next, please let me know. Should I rebase on the current main to pass this test? (And then keep checking if I need to rebase again as I don't know when the tests will next be run..?)

File: internal/service/ssoadmin/customer_managed_policy_attachment.go
Comment (bschaatsbergen) [inline internal/service/ssoadmin/customer_managed_policy_attachment.go:55]: As per https://docs.aws.amazon.com/singlesignon/latest/APIReference/API_CustomerManagedPolicyReference.html

```suggestion
							ForceNew: true,
							ValidateFunc: validation.StringLenBetween(0, 128),
```

File: internal/service/ssoadmin/customer_managed_policy_attachment.go
Comment (bschaatsbergen) [inline internal/service/ssoadmin/customer_managed_policy_attachment.go:61]: As per https://docs.aws.amazon.com/singlesignon/latest/APIReference/API_CustomerManagedPolicyReference.html
```suggestion
							ForceNew: true,
							ValidateFunc: validation.StringLenBetween(0, 512),
```

File: internal/service/ssoadmin/customer_managed_policy_attachment.go
Comment (bschaatsbergen) [inline internal/service/ssoadmin/customer_managed_policy_attachment.go:110]: Could you please rephrase this comment a bit to in order to make it more understandable what actually needs to happen

File: internal/service/ssoadmin/customer_managed_policy_attachment.go
Comment (bschaatsbergen) [inline internal/service/ssoadmin/customer_managed_policy_attachment.go:167]: Perhaps we can annotate the below logic with a comment to explain why we're retrying.. I believe it's due to IAM eventual consistency?

File: internal/service/ssoadmin/customer_managed_policy_attachment.go
Comment (bschaatsbergen) [inline internal/service/ssoadmin/customer_managed_policy_attachment.go:195]: Same comment as earlier, lets be a bit more specific on what/why we're doing exactly here.

File: internal/service/ssoadmin/customer_managed_policy_attachment_test.go
Comment (bschaatsbergen) [inline internal/service/ssoadmin/customer_managed_policy_attachment_test.go:5]: Should/can we remove this if it's not used?

File: website/docs/r/ssoadmin_customer_managed_policy_attachment.html.markdown
Comment (bschaatsbergen) [inline website/docs/r/ssoadmin_customer_managed_policy_attachment.html.markdown:49]: Please lint the fenced Terraform code block

**Review by bschaatsbergen — COMMENTED** [review]
Overall the contribution looks great, I've left a few comments here and there to make things a bit more clear. 

Regarding tests; I seem to be unable to run the tests as I'm getting an access denied exception (something I have to sort out).

I believe that @ewbankkit will perform an additional review before we merge this.

**Comment by oakbramble** [comment]
Thanks for the comments @bschaatsbergen :relaxed:  I have updated everything you mentioned. If you have any further feedback, just let me know.

**Comment by oakbramble** [comment]
Ok, the semgrep check failed on the last run. I saw that several semgrep changes had been merged recently and it looks like that was the cause of the failure. I have rebased again to remedy this. As ever, if you think that isn't the case and I have missed something, please let me know.

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccSSOAdminCustomerManagedPolicyAttachment_' PKG=ssoadmin ACCTEST_PARALLELISM=2 
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/ssoadmin/... -v -count 1 -parallel 2  -run=TestAccSSOAdminCustomerManagedPolicyAttachment_ -timeout 180m
=== RUN   TestAccSSOAdminCustomerManagedPolicyAttachment_basic
=== PAUSE TestAccSSOAdminCustomerManagedPolicyAttachment_basic
=== RUN   TestAccSSOAdminCustomerManagedPolicyAttachment_forceNew
=== PAUSE TestAccSSOAdminCustomerManagedPolicyAttachment_forceNew
=== RUN   TestAccSSOAdminCustomerManagedPolicyAttachment_disappears
--- PASS: TestAccSSOAdminCustomerManagedPolicyAttachment_disappears (29.85s)
=== RUN   TestAccSSOAdminCustomerManagedPolicyAttachment_Disappears_permissionSet
--- PASS: TestAccSSOAdminCustomerManagedPolicyAttachment_Disappears_permissionSet (20.95s)
=== RUN   TestAccSSOAdminCustomerManagedPolicyAttachment_multipleManagedPolicies
=== PAUSE TestAccSSOAdminCustomerManagedPolicyAttachment_multipleManagedPolicies
=== CONT  TestAccSSOAdminCustomerManagedPolicyAttachment_basic
=== CONT  TestAccSSOAdminCustomerManagedPolicyAttachment_multipleManagedPolicies
--- PASS: TestAccSSOAdminCustomerManagedPolicyAttachment_basic (32.87s)
=== CONT  TestAccSSOAdminCustomerManagedPolicyAttachment_forceNew
--- PASS: TestAccSSOAdminCustomerManagedPolicyAttachment_multipleManagedPolicies (52.84s)
--- PASS: TestAccSSOAdminCustomerManagedPolicyAttachment_forceNew (56.37s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/ssoadmin	143.892s
```

**Pull request merged**
